ff-doc-library/projects/ff-doc/src/lib/react/javadoc-transform.hook.ts

```diff
@@ -0,0 +1,55 @@
+import { LinkData, transformAsHtml, transformAsText } from '../javadoc';
+import { ElementClass } from '../frankdoc.types';
+import { useMemo } from 'react';
+
+type InnerHTML = {
+  __html: string;
+};
+
+function javadocTransform(
+  javadoc: string | undefined,
+  elements: Record<string, ElementClass> | null,
+  asText = true,
+  // eslint-disable-next-line no-unused-vars
+  linkTemplate?: (link: LinkData) => string,
+): string {
+  if (javadoc === '') javadoc = '-';
+  if (!javadoc || !elements) return '';
+
+  const hasCustomLinkTransform = !!linkTemplate;
+  const javadocParts = asText
+    ? transformAsText(javadoc, elements)
+    : transformAsHtml(javadoc, elements, hasCustomLinkTransform);
+
+  let transformedResult = '';
+  for (const partIndexString in javadocParts) {
+    const partIndex = +partIndexString;
+    const part = javadocParts[partIndex];
+    if (hasCustomLinkTransform && partIndex % 2 !== 0 && part.startsWith('{')) {
```

Comment:
This line makes me realize that I don't fully understand the code. Could it be useful to clarify why only odd items can be parsed? I'm not sure what `transformAsText` and `transformAsHtml` produce that this needs to be done.

Maybe adding a test for it would be nice, as to provide an example of the output.